core/slim_sim.cpp

```diff
@@ -4877,30 +4877,49 @@ void SLiMSim::WriteIndividualTable(table_collection_t *p_tables)
 {
     // This overwrites whatever might be previously in the individual table
     individual_table_clear(p_tables->individuals);
+    // First prepend the existing remembered individuals
+    individual_table_copy(tables.individuals, p_tables->individuals);
 	
+    // TODO: CHECK THIS IS UNNEEDED?!?
     // reset the individual column to -1 (which is MSP_NULL_INDIVIDUAL)
-    memset(p_tables->nodes->individual, 0xff, p_tables->nodes->num_rows * sizeof(individual_id_t));
+    // memset(p_tables->nodes->individual, 0xff, p_tables->nodes->num_rows * sizeof(individual_id_t));
```

Comment:
> I'm not sure why the reset to -1 for the individual column would be unneeded now. Simplify may not throw away individual table entries any more if the corresponding nodes are gone; but the same is not true for SLiM, right? There can be nodes in the tree sequence for which (a) SLiM has forgotten the individual because it is dead and gone, and (b) it was not a "remembered individual", so there's no saved individual table entry for it, right? So won't those nodes still need to be given -1?


Ah, this is important. We don't want to recompute `tables.nodes.individual`each time around. But, correct me if I am wrong, but I think this is OK. Simplify *does* throw away individuals with no remaining nodes; but these individual's nodes are guaranteed to be around still. The only individuals kept in `tables.individuals` are Remembered ones -- others get added to the copy of the tables that get output, but that procedure doesn't touch the underlying `tables` that get carried forward.

Reply:
I suspect I'm missing the point somehow.  But p_tables is the output table, and contains all of the nodes in the current tree sequence.  The job of this method is to set up individual-related information in p_tables, not just for the remembered and extant individuals, but also for all of the nodes in p_tables.  Some of the nodes in p_tables will correspond to individuals that were not remembered and are no longer extant, and so the p_tables->nodes->individual entries for those nodes need to have -1 set for their individual.  So the strategy up until now, as I understand it, has been to set p_tables->nodes->individual to all -1, and then to fill in the right values for those nodes that still correspond to an individual in the table.  The only way that setting to -1 would be unnecessary would be if every entry in p_tables->nodes corresponded to an entry in the individuals table anyway, in which case every -1 value would end up being overwritten in the loop below.  But that is not the case here, as I understand it.  No?  Am I being dense?  Maybe we need to talk about this in person.